### DISCUSSION TITLE
Eye texture paint, UV, mapping?

### DISCUSSION THREAD

**Discussion by Sorry-Im-Stupid** [bnpr/Malt#119] [Q&A]
I've been making characters to test my texture work in Malt, and the skin texture looks good, works quite nicely on a humanoid model. But now I'm working on the eyes, and I can't figure out how to get the eye texture to show up properly in the model. I tried with screen UV, and of course it only works from w/e camera angle you paint it in.

I can't find a node that works the way it would in cycles and eevee, to assign the particular image to a set of faces as set by the UV map.
I'm going to try doing it as a relative set with the rings of the eye model, but that limits the amount of detail I can achieve, so an image-based solution would be ideal.

This is what I've got so far from that method. As you can see, not ideal.
![Eye test0](https://user-images.githubusercontent.com/84992987/131885739-3909467c-35dc-43dc-9397-0bddd5ff2df9.png)

**Comment by pragma37** [comment]
There's a surface_uv node in Node Utils - common.

![imagen](https://user-images.githubusercontent.com/36610999/132026012-bb61e856-e8bc-4f11-8903-82f35da9e55b.png)

**Reply by Sorry-Im-Stupid** [reply]
I can't find that anywhere. It's not in the one I've been using, and I can't even find the malt node system in the latest release.

**Reply by pragma37** [reply]
You were using the version I made to workaround shader compilations issues on your Linux distribution, right?
I think I added the surface nodes later.

You can also access UVs like this:
![imagen](https://user-images.githubusercontent.com/36610999/132043588-013d83d2-c87a-4cce-9797-c43b1e20bb14.png)

You could also use the inline code node:
![imagen](https://user-images.githubusercontent.com/36610999/132044022-8fdc456a-89b5-427d-ac01-5655d0b10693.png)

**Reply by Sorry-Im-Stupid** [reply]
I am using that, yes. But the new one, the Malt Node Editor isn't even in the dropdown menu.

**Reply by pragma37** [reply]
You probably downloaded the Release version, only the Development version has nodes.
Anyway, if shader compilation doesn't work for you then it doesn't really matter.

**Reply by Sorry-Im-Stupid** [reply]
Actually, now that you mention it, it didn't spit out a compilation error. Maybe I didn't get that far. I'll look for the dev version and get back to you.

**Reply by Sorry-Im-Stupid** [reply]
Yep. Compiler errors.
